akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -154,6 +156,12 @@ object Sink extends SinkApply {
           promise.success(())
           ctx.finish()
         }
+
+        override def decide(cause: Throwable): Supervision.Directive = {
+          // supervision will be implemented by #16916
+          promise.tryFailure(cause)
+          super.decide(cause)
```

Comment:
I guess that means that #16916 will allow the supervision to decide if the promise is completed with the failure?